### PR TITLE
ci: add workflow for releases

### PR DESCRIPTION
**Pull request by milas**
GitHub Actions workflow to create a release: will upload to PyPI
and create a GitHub release with the `sdist` and `bdist_wheel`
as well.

The version code is switched to `setuptools_scm` to work well
with this flow (e.g. avoid needing to write a script that does
a `sed` on the version file and commits as part of release).

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "Release Tag (e.g. 6.0.0)"
```
Comment (nicksieger) [inline .github/workflows/release.yml:7]: Suggest putting the "v" in the example to make it clear what the tag will be

```suggestion
        description: "Release Tag (e.g. v6.0.0)"
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
```
Comment (nicksieger) [inline .github/workflows/release.yml:5]: What do you think of adding a "dry-run" flag for testing the release, like in [compose-cli](https://github.com/docker/compose-cli/blob/main/.github/workflows/cli-release.yaml#L9)?

**Review by nicksieger — CHANGES_REQUESTED** [review]
Mostly LGTM, the setuptools_scm stuff is opaque so I assume you've got it right and it's working

File: .github/workflows/release.yml (hunk 1 above)
Comment (nicksieger) [reply .github/workflows/release.yml:7]: Nevermind, I see that the existing convention is w/o the `v` prefix

**Comment by milas** [comment]
> the setuptools_scm stuff is opaque to me so I assume you've got it right and it's working

Ha yeah everything is pretty much lifted straight from their docs! It's funky but less crazy than most other solutions I've seen (and is "blessed" by pypa that maintains pip etc)

File: .github/workflows/release.yml (hunk 1 above)
Comment (milas) [reply .github/workflows/release.yml:7]: Yes! I'll make this clearer in the description, that was why I gave the example w/o `v` - Python versions/tags do NOT want it as a prefix 😬

File: .github/workflows/release.yml (hunk 2 above)
Comment (milas) [reply .github/workflows/release.yml:5]: Fancy! I stole it and made it fancier (✨ types ✨)

**Review by nicksieger — APPROVED** [review]

**Pull request merged**